Likes: make module compatible with WP.com Simple sites

jsnajdr opened the pull request:
Several little changes in Likes module that let it run smoothly in WP.com Simple site environment, which has its own very special Jetpack loader.

First, the `Jetpack::enable_module_configurable` method call is conditional, after checking that the class and the method really exists. WP.com has its own implementation of the `Jetpack` class (`class.jetpack.php`) which doesn't have the enable_module_configurable` method. The same `$this->in_jetpack` condition is already present in two other Jetpack modules: `publicize.php` and `jetpack-carousel.php`. Here we do the same thing in a third module.

Second, the WP.com implementation of the `Jetpack` class doesn't have the `get_active_modules` method. It only has a `is_module_active` method that always returns `true`, for all modules. I'm starting to use it also in the Likes module. It was always a better idea to use `is_module_active` instead or a combination of `get_active_modules` and `in_array`.

## Does this pull request change what data or activity we track or use?

No

## Testing instructions:

* Verify that Likes still work as they worked before and that the two variants of the "is active" check are equivalent for all practical purposes.

jsnajdr committed:
Likes: make module compatible with WP.com Simple sites

tyxla commented on projects/plugins/jetpack/modules/likes.php:
```diff
@@ -40,6 +40,12 @@
  */
 class Jetpack_Likes {
+	/**
+	 * Determines whether the Jetpack class and method exists. Default is true.
```
Does it make sense to say "determines whether we're running/executing inside the Jetpack plugin"? I think it conveys our intention better.

tyxla commented on projects/plugins/jetpack/modules/likes.php:
```diff
@@ -65,6 +71,8 @@ public static function init() {
 	 * Constructs Likes class
 	 */
 	public function __construct() {
+		$this->in_jetpack = class_exists( 'Jetpack' ) && method_exists( 'Jetpack', 'enable_module_configurable' );
```
I would have loved if we had a utility function to do this consistently already, but seems like we don't. Plus, this is being done the same way in the publicize module, so I'm fine to do it the same way.

tyxla commented on projects/plugins/jetpack/modules/likes.php:
```diff
@@ -76,20 +84,24 @@ public function __construct() {
 		add_filter( 'sharing_show_buttons_on_row_start', array( $this, 'configuration_target_area' ) );
 
-		$active = Jetpack::get_active_modules();
+		$publicize_active  = Jetpack::is_module_active( 'publicize' );
+		$sharedaddy_active = Jetpack::is_module_active( 'sharedaddy' );
```
The `is_module_active()` makes things more readable, thanks!

However, won't these `Jetpack::` static calls die if the `Jetpack` class doesn't exist? Should we wrap them in a `$this->in_jetpack` check?

anomiex replied:
The "in_jetpack" seems like a bit of a misnomer to me here. Unless we're planning on splitting this out into a separate package for some reason, the real distinction instead seems to be "is the `Jetpack` class the normal one or Simple's stubby version?".

anomiex commented on projects/plugins/jetpack/modules/likes.php:
```diff
@@ -76,20 +84,24 @@ public function __construct() {
 
-		Jetpack::enable_module_configurable( __FILE__ );
+		if ( $this->in_jetpack ) {
+			Jetpack::enable_module_configurable( __FILE__ );
```
An alternative would be to add this function to Simple's stub version of the Jetpack class. Or do a somewhat more standard check for Simple (`defined( 'IS_WPCOM' ) && IS_WPCOM`, or use `Automattic\Jetpack\Status\Host` like `( new Host() )->is_wpcom_simple()`).

jsnajdr committed:
Use host->is_wpcom_simple consistently

jsnajdr commented:
Thanks for your comments @anomiex and @tyxla. Here's how I addressed your feedback:
- the existing code that checks for existence of `enable_module_configurable` is all 13 years old, from 2012. Today we have more modern methods to do that check, namely the `$host->is_wpcom_simple()` method. I updated all usages to use this method.
- we can add `Jetpack::enable_module_configurable` to the WP.com stub, but only after the deployed Jetpack version stops checking for its existence and making conlusions from the result. This PR is a step towards that, there are no longer any `method_exists` checks.
- the `in_jetpack` name may be a misnomer (would `in_simple` be a better name, with a flipped boolean value?) but it's used quite consistently across several files, so I decided to keep it, at least for now.
- won't `Jetpack::` static calls die if the `Jetpack` class doesn't exist? Yes, they will, but we're never guarding for that. The Jetpack codebase has hundreds of such static calls. In all supported environments, the class always exists, it only has different implementations.

jsnajdr commented on projects/plugins/jetpack/modules/publicize.php:
```diff
@@ -17,34 +17,19 @@
-	 *
-	 * @var Automattic\Jetpack\Modules
-	 */
-	public $modules;
```
I noticed that the `$this->modules` field is no longer used, so I'm removing it. Existing usages were gradually removed (in PRs like #35781) and eventually usage dropped to zero, unnoticed.

jsnajdr committed:
Update phan baselines

tyxla reviewed (approved):
This is looking good from my perspective 👍 

Thanks for the drive-by consistency improvements @jsnajdr 🚀
